I still get the following indentation errors from `./gradlew lint`

```
[ant:checkstyle] [ERROR] C:\Users\Lucas\courses\sef18\teammates\src\main\java\teammates\common\datatransfer\questions\FeedbackNumericalScaleQuestionDetails.java:27: 'FeedbackQuestionDetails' has incorrect indentation level 0, expected level should be 8. [Indentation]
[ant:checkstyle] [ERROR] C:\Users\Lucas\courses\sef18\teammates\src\main\java\teammates\common\datatransfer\questions\FeedbackNumericalScaleQuestionDetails.java:50: 'HttpRequestHelper' has incorrect indentation level 12, expected level should be 16. [Indentation]
[ant:checkstyle] [ERROR] C:\Users\Lucas\courses\sef18\teammates\src\main\java\teammates\common\datatransfer\questions\FeedbackNumericalScaleQuestionDetails.java:56: 'HttpRequestHelper' has incorrect indentation level 12, expected level should be 16. [Indentation]
[ant:checkstyle] [ERROR] C:\Users\Lucas\courses\sef18\teammates\src\main\java\teammates\common\datatransfer\questions\FeedbackNumericalScaleQuestionDetails.java:62: 'HttpRequestHelper' has incorrect indentation level 12, expected level should be 16. [Indentation]
[ant:checkstyle] [ERROR] C:\Users\Lucas\courses\sef18\teammates\src\main\java\teammates\common\datatransfer\questions\FeedbackNumericalScaleQuestionDetails.java:88: '(' has incorrect indentation level 12, expected level should be 16. [Indentation]
[ant:checkstyle] [ERROR] C:\Users\Lucas\courses\sef18\teammates\src\main\java\teammates\common\datatransfer\questions\FeedbackNumericalScaleQuestionDetails.java:152: 'method def rcurly' has incorrect indentation level 8, expected level should be 4. [Indentation]
[ant:checkstyle] [ERROR] C:\Users\Lucas\courses\sef18\teammates\src\main\java\teammates\common\datatransfer\questions\FeedbackNumericalScaleQuestionDetails.java:158: '+' has incorrect indentation level 12, expected level should be 16. [Indentation]
[ant:checkstyle] [ERROR] C:\Users\Lucas\courses\sef18\teammates\src\main\java\teammates\common\datatransfer\questions\FeedbackNumericalScaleQuestionDetails.java:159: '+' has incorrect indentation level 12, expected level should be 16. [Indentation]
[ant:checkstyle] [ERROR] C:\Users\Lucas\courses\sef18\teammates\src\main\java\teammates\common\datatransfer\questions\FeedbackNumericalScaleQuestionDetails.java:160: '+' has incorrect indentation level 12, expected level should be 16. [Indentation]
[ant:checkstyle] [ERROR] C:\Users\Lucas\courses\sef18\teammates\src\main\java\teammates\common\datatransfer\questions\FeedbackNumericalScaleQuestionDetails.java:206: '?' has incorrect indentation level 12, expected level should be 16. [Indentation]
[ant:checkstyle] [ERROR] C:\Users\Lucas\courses\sef18\teammates\src\main\java\teammates\common\datatransfer\questions\FeedbackNumericalScaleQuestionDetails.java:207: ':' has incorrect indentation level 12, expected level should be 16. [Indentation]
[ant:checkstyle] [ERROR] C:\Users\Lucas\courses\sef18\teammates\src\main\java\teammates\common\datatransfer\questions\FeedbackNumericalScaleQuestionDetails.java:221: 'getAverageExcludingSelfText' has incorrect indentation level 16, expected level should be 20. [Indentation]
[ant:checkstyle] [ERROR] C:\Users\Lucas\courses\sef18\teammates\src\main\java\teammates\common\datatransfer\questions\FeedbackNumericalScaleQuestionDetails.java:242: '?' has incorrect indentation level 12, expected level should be 16. [Indentation]
[ant:checkstyle] [ERROR] C:\Users\Lucas\courses\sef18\teammates\src\main\java\teammates\common\datatransfer\questions\FeedbackNumericalScaleQuestionDetails.java:243: ':' has incorrect indentation level 12, expected level should be 16. [Indentation]
[ant:checkstyle] [ERROR] C:\Users\Lucas\courses\sef18\teammates\src\main\java\teammates\common\datatransfer\questions\FeedbackNumericalScaleQuestionDetails.java:273: '?' has incorrect indentation level 12, expected level should be 16. [Indentation]
[ant:checkstyle] [ERROR] C:\Users\Lucas\courses\sef18\teammates\src\main\java\teammates\common\datatransfer\questions\FeedbackNumericalScaleQuestionDetails.java:274: ':' has incorrect indentation level 12, expected level should be 16. [Indentation]
[ant:checkstyle] [ERROR] C:\Users\Lucas\courses\sef18\teammates\src\main\java\teammates\common\datatransfer\questions\FeedbackNumericalScaleQuestionDetails.java:280: '||' has incorrect indentation level 12, expected level should be 16. [Indentation]
[ant:checkstyle] [ERROR] C:\Users\Lucas\courses\sef18\teammates\src\main\java\teammates\common\datatransfer\questions\FeedbackNumericalScaleQuestionDetails.java:304: '?' has incorrect indentation level 12, expected level should be 16. [Indentation]
[ant:checkstyle] [ERROR] C:\Users\Lucas\courses\sef18\teammates\src\main\java\teammates\common\datatransfer\questions\FeedbackNumericalScaleQuestionDetails.java:305: ':' has incorrect indentation level 12, expected level should be 16. [Indentation]
[ant:checkstyle] [ERROR] C:\Users\Lucas\courses\sef18\teammates\src\main\java\teammates\common\datatransfer\questions\FeedbackNumericalScaleQuestionDetails.java:340: 'getAverageExcludingSelfText' has incorrect indentation level 12, expected level should be 16. [Indentation]
[ant:checkstyle] [ERROR] C:\Users\Lucas\courses\sef18\teammates\src\main\java\teammates\common\datatransfer\questions\FeedbackNumericalScaleQuestionDetails.java:449: '+' has incorrect indentation level 12, expected level should be 16. [Indentation]
[ant:checkstyle] [ERROR] C:\Users\Lucas\courses\sef18\teammates\src\main\java\teammates\common\datatransfer\questions\FeedbackNumericalScaleQuestionDetails.java:450: '+' has incorrect indentation level 12, expected level should be 16. [Indentation]
[ant:checkstyle] [ERROR] C:\Users\Lucas\courses\sef18\teammates\src\main\java\teammates\common\datatransfer\questions\FeedbackNumericalScaleQuestionDetails.java:461: 'method def' child has incorrect indentation level 12, expected level should be 8. [Indentation]
[ant:checkstyle] [ERROR] C:\Users\Lucas\courses\sef18\teammates\src\main\java\teammates\common\datatransfer\questions\FeedbackNumericalScaleQuestionDetails.java:462: 'method def rcurly' has incorrect indentation level 8, expected level should be 4. [Indentation]
[ant:checkstyle] [ERROR] C:\Users\Lucas\courses\sef18\teammates\src\main\java\teammates\common\datatransfer\questions\FeedbackNumericalScaleQuestionDetails.java:478: 'getAverageExcludingSelfText' has incorrect indentation level 12, expected level should be 16. [Indentation]
[ant:checkstyle] [ERROR] C:\Users\Lucas\courses\sef18\teammates\src\main\java\teammates\common\datatransfer\questions\FeedbackNumericalScaleQuestionDetails.java:582: '(' has incorrect indentation level 16, expected level should be 20. [Indentation]
[ant:checkstyle] [ERROR] C:\Users\Lucas\courses\sef18\teammates\src\main\java\teammates\common\datatransfer\questions\FeedbackNumericalScaleQuestionDetails.java:694: '(' has incorrect indentation level 12, expected level should be 16. [Indentation]
[ant:checkstyle] [ERROR] C:\Users\Lucas\courses\sef18\teammates\src\main\java\teammates\common\datatransfer\questions\FeedbackNumericalScaleQuestionDetails.java:767: '(' has incorrect indentation level 16, expected level should be 20. [Indentation]
```

--- a/src/main/java/teammates/common/datatransfer/questions/FeedbackNumericalScaleQuestionDetails.java
+++ b/src/main/java/teammates/common/datatransfer/questions/FeedbackNumericalScaleQuestionDetails.java
@@ -24,7 +24,7 @@ import teammates.common.util.Templates.FeedbackQuestion.Slots;
 import teammates.ui.template.InstructorFeedbackResultsResponseRow;
 
 public class FeedbackNumericalScaleQuestionDetails extends
-FeedbackQuestionDetails {
+        FeedbackQuestionDetails {
     private int minScale;
     private int maxScale;
     private double step;
@@ -47,19 +47,19 @@ FeedbackQuestionDetails {
             FeedbackQuestionType questionType) {
 
         String minScaleString =
-            HttpRequestHelper.getValueFromParamMap(requestParameters,
+                HttpRequestHelper.getValueFromParamMap(requestParameters,
                     Const.ParamsNames.FEEDBACK_QUESTION_NUMSCALE_MIN);
         Assumption.assertNotNull("Null minimum scale", minScaleString);
         int minScale = Integer.parseInt(minScaleString);
 
         String maxScaleString =
-            HttpRequestHelper.getValueFromParamMap(requestParameters,
+                HttpRequestHelper.getValueFromParamMap(requestParameters,
                     Const.ParamsNames.FEEDBACK_QUESTION_NUMSCALE_MAX);
         Assumption.assertNotNull("Null maximum scale", maxScaleString);
         int maxScale = Integer.parseInt(maxScaleString);
 
         String stepString =
-            HttpRequestHelper.getValueFromParamMap(requestParameters,
+                HttpRequestHelper.getValueFromParamMap(requestParameters,
                     Const.ParamsNames.FEEDBACK_QUESTION_NUMSCALE_STEP);
         Assumption.assertNotNull("Null step", stepString);
         Double step = Double.parseDouble(stepString);
@@ -82,10 +82,10 @@ FeedbackQuestionDetails {
 
     @Override
     public String getQuestionWithExistingResponseSubmissionFormHtml(
-            boolean sessionIsOpen, int qnIdx, int responseIdx, String courseId,
-            int totalNumRecipients, FeedbackResponseDetails existingResponseDetails) {
+                boolean sessionIsOpen, int qnIdx, int responseIdx, String courseId,
+                int totalNumRecipients, FeedbackResponseDetails existingResponseDetails) {
         FeedbackNumericalScaleResponseDetails numscaleResponseDetails =
-            (FeedbackNumericalScaleResponseDetails) existingResponseDetails;
+                (FeedbackNumericalScaleResponseDetails) existingResponseDetails;
 
         return Templates.populateTemplate(
                 FormTemplates.NUMSCALE_SUBMISSION_FORM,
@@ -149,15 +149,15 @@ FeedbackQuestionDetails {
         return "<div id=\"numScaleForm\">"
             + this.getQuestionSpecificEditFormHtml(-1)
             + "</div>";
-        }
+    }
 
     @Override
     public String getQuestionAdditionalInfoHtml(int questionNumber,
             String additionalInfoId) {
         String additionalInfo = getQuestionTypeDisplayName()
-            + ":<br>Minimum value: " + minScale
-            + ". Increment: " + step + ". Maximum value: "
-            + maxScale + '.';
+                + ":<br>Minimum value: " + minScale
+                + ". Increment: " + step + ". Maximum value: "
+                + maxScale + '.';
 
         return Templates.populateTemplate(
                 FormTemplates.FEEDBACK_QUESTION_ADDITIONAL_INFO,
@@ -203,8 +203,8 @@ FeedbackQuestionDetails {
         boolean showAvgExcludingSelf = showAverageExcludingSelf(question, averageExcludingSelf);
 
         String fragmentTemplateToUse = showAvgExcludingSelf
-            ? FormTemplates.NUMSCALE_RESULTS_STATS_FRAGMENT_WITH_SELF_RESPONSE
-            : FormTemplates.NUMSCALE_RESULTS_STATS_FRAGMENT;
+                ? FormTemplates.NUMSCALE_RESULTS_STATS_FRAGMENT_WITH_SELF_RESPONSE
+                : FormTemplates.NUMSCALE_RESULTS_STATS_FRAGMENT;
 
         DecimalFormat df = customDecimalFormat(0, 5, "down");
 
@@ -218,7 +218,7 @@ FeedbackQuestionDetails {
 
             Double userAverageExcludingSelf = averageExcludingSelf.get(recipient);
             String userAverageExcludingSelfText =
-                getAverageExcludingSelfText(showAvgExcludingSelf, df, userAverageExcludingSelf);
+                    getAverageExcludingSelfText(showAvgExcludingSelf, df, userAverageExcludingSelf);
 
             String recipientName = recipient.equals(Const.GENERAL_QUESTION) ? "General" : bundle.getNameForEmail(recipient);
             String recipientTeam = bundle.getTeamNameForEmail(recipient);
@@ -239,8 +239,8 @@ FeedbackQuestionDetails {
 
         String statsTitle = "Response Summary";
         String templateToUse = showAvgExcludingSelf
-            ? FormTemplates.NUMSCALE_RESULT_STATS_WITH_SELF_RESPONSE
-            : FormTemplates.NUMSCALE_RESULT_STATS;
+                ? FormTemplates.NUMSCALE_RESULT_STATS_WITH_SELF_RESPONSE
+                : FormTemplates.NUMSCALE_RESULT_STATS;
         return Templates.populateTemplate(
                 templateToUse,
                 Slots.SUMMARY_TITLE, statsTitle,
@@ -270,14 +270,14 @@ FeedbackQuestionDetails {
         boolean showAvgExcludingSelf = showAverageExcludingSelf(question, averageExcludingSelf);
 
         String fragmentTemplateToUse = showAvgExcludingSelf
-            ? FormTemplates.NUMSCALE_RESULTS_STATS_FRAGMENT_WITH_SELF_RESPONSE
-            : FormTemplates.NUMSCALE_RESULTS_STATS_FRAGMENT;
+                ? FormTemplates.NUMSCALE_RESULTS_STATS_FRAGMENT_WITH_SELF_RESPONSE
+                : FormTemplates.NUMSCALE_RESULTS_STATS_FRAGMENT;
 
         DecimalFormat df = customDecimalFormat(0, 5, "down");
 
         boolean isRecipientTypeGeneral = question.recipientType == FeedbackParticipantType.NONE;
         boolean isRecipientTypeTeam = question.recipientType == FeedbackParticipantType.TEAMS
-            || question.recipientType == FeedbackParticipantType.OWN_TEAM;
+                || question.recipientType == FeedbackParticipantType.OWN_TEAM;
         boolean isRecipientTypeStudent = !isRecipientTypeGeneral && !isRecipientTypeTeam;
 
         String currentUserTeam = bundle.getTeamNameForEmail(studentEmail);
@@ -301,14 +301,15 @@ FeedbackQuestionDetails {
                 isRecipientTypeGeneral, isRecipientTypeTeam,
                 hasAtLeastTwoResponsesOtherThanCurrentUser(numResponses, currentUserIdentifier, hiddenRecipients));
         String templateToUse = showAvgExcludingSelf
-            ? FormTemplates.NUMSCALE_RESULT_STATS_WITH_SELF_RESPONSE
-            : FormTemplates.NUMSCALE_RESULT_STATS;
+                ? FormTemplates.NUMSCALE_RESULT_STATS_WITH_SELF_RESPONSE
+                : FormTemplates.NUMSCALE_RESULT_STATS;
         return Templates.populateTemplate(templateToUse,
                 Slots.SUMMARY_TITLE, statsTitle,
                 Slots.STATS_FRAGMENTS, fragmentHtml.toString());
     }
 
-    private void UpdateRecipientList(ArrayList<String> recipientList, Set<String> recipientSet, String currentUserIdentifier) {
+    private void updateRecipientList(ArrayList<String> recipientList,
+            Set<String> recipientSet, String currentUserIdentifier) {
         boolean hasCurrentUserReceivedAnyResponse = recipientSet.contains(currentUserIdentifier);
 
         // Move current user to the head of the recipient list
@@ -324,9 +325,10 @@ FeedbackQuestionDetails {
             recipientList.add(otherRecipient);
         }
     }
-    /** update a template with one single record
+    /** update a template with one single record.
      * @return updated template
-     */
+         */
+    
     private String updateTemplate(String currentTemplate, Map<String, Double> min, Map<String, Double> max,
             Map<String, Double> average, Map<String, Double> averageExcludingSelf, String recipient, 
             DecimalFormat df, String recipientTeam, String recipientName, boolean showAvgExcludingSelf) {
@@ -337,7 +339,7 @@ FeedbackQuestionDetails {
         Double averageScoreExcludingSelf = averageExcludingSelf.get(recipient);
 
         String averageScoreExcludingSelfText =
-            getAverageExcludingSelfText(showAvgExcludingSelf, df, averageScoreExcludingSelf);
+                getAverageExcludingSelfText(showAvgExcludingSelf, df, averageScoreExcludingSelf);
 
         String newTemplate = Templates.populateTemplate(
                 currentTemplate,
@@ -446,8 +448,8 @@ FeedbackQuestionDetails {
         DecimalFormat df = customDecimalFormat(0, 5, "down");
 
         String csvHeader = "Team, Recipient, Average, Minimum, Maximum"
-            + (showAvgExcludingSelf ? ", Average excluding self response" : "")
-            + Const.EOL;
+                + (showAvgExcludingSelf ? ", Average excluding self response" : "")
+                + Const.EOL;
 
         StringBuilder csvBody = new StringBuilder();
         for (String recipient : numResponses.keySet()) {
@@ -455,13 +457,14 @@ FeedbackQuestionDetails {
             if (hiddenRecipients.contains(recipient)) {
                 continue;
             }
-            updateQuestionResultStatisticsCsv(csvBody, bundle, recipient, average, min, max, showAvgExcludingSelf, averageExcludingSelf);
+            updateQuestionResultStatisticsCsv(csvBody, bundle, recipient, average, 
+                    min, max, showAvgExcludingSelf, averageExcludingSelf);
         }
 
-            return csvHeader + csvBody.toString();
-        }
+        return csvHeader + csvBody.toString();
+    }
 
-    /** adds a single record to the QuestionResultStatisticsCsv
+    /** adds a single record to the QuestionResultStatisticsCsv.
      * @return void
      */
     private void updateQuestionResultStatisticsCsv(StringBuilder csvBody, FeedbackSessionResultsBundle bundle,
@@ -475,7 +478,7 @@ FeedbackQuestionDetails {
 
         Double averageScoreExcludingSelf = averageExcludingSelf.get(recipient);
         String averageScoreExcludingSelfText =
-            getAverageExcludingSelfText(showAvgExcludingSelf, df, averageScoreExcludingSelf);
+                getAverageExcludingSelfText(showAvgExcludingSelf, df, averageScoreExcludingSelf);
 
         csvBody.append(SanitizationHelper.sanitizeForCsv(recipientTeam) + ','
                 + SanitizationHelper.sanitizeForCsv(isRecipientGeneral
@@ -489,7 +492,7 @@ FeedbackQuestionDetails {
                 + Const.EOL);
     }
 
-    /** creates a DecimalFormat with a given minimum & maximum fraction digits, and a rounding mode
+    /** creates a DecimalFormat with a given minimum & maximum fraction digits, and a rounding mode.
      * @return DecimalFormat
      */
     private DecimalFormat customDecimalFormat(int min, int max, String roundingMode) {
@@ -497,9 +500,9 @@ FeedbackQuestionDetails {
         df.setMinimumFractionDigits(min);
         df.setMaximumFractionDigits(max);
         //  roundingMode = roundingMode.toUpperCase()
-        if (roundingMode.equals("down")) {
+        if (roundingMode.equals(new String("down"))) {
             df.setRoundingMode(RoundingMode.DOWN);
-        } else if (roundingMode.equals("up")) {
+        } else if (roundingMode.equals(new String("up"))) {
             df.setRoundingMode(RoundingMode.UP);
         }
         return df;
@@ -512,10 +515,9 @@ FeedbackQuestionDetails {
             String currentUserTeam, String currentUserIdentifier, String fragmentTemplateToUse, 
             FeedbackSessionResultsBundle bundle, Map<String, Double> averageExcludingSelf,
             Map<String, Integer> numResponses, Map<String, Double> min, Map<String, Double> max,
-            Map<String, Double> average, DecimalFormat df)
-    {
+            Map<String, Double> average, DecimalFormat df) {
 
-        UpdateRecipientList(recipientList, recipientSet, currentUserIdentifier);
+        updateRecipientList(recipientList, recipientSet, currentUserIdentifier);
         StringBuilder fragmentHtml = new StringBuilder(); 
         for (String recipient : recipientList) {                                                                    
 
@@ -534,17 +536,19 @@ FeedbackQuestionDetails {
                     hasAtLeastTwoResponsesOtherThanCurrentUser(numResponses, currentUserIdentifier, hiddenRecipients), 
                     isRecipientGeneral, bundle.getNameForEmail(recipient), currentUserTeam);                
 
-            String recipientTeam = getDisplayableRecipientTeam(isHiddenRecipient, isRecipientCurrentUser, hasAtLeastTwoResponses(numResponses, currentUserIdentifier),
+            String recipientTeam = getDisplayableRecipientTeam(isHiddenRecipient, isRecipientCurrentUser,
+                    hasAtLeastTwoResponses(numResponses, currentUserIdentifier),
                     isRecipientTypeStudent, hasAtLeastTwoResponsesOtherThanCurrentUser(numResponses,
                         currentUserIdentifier, hiddenRecipients),
                     bundle.getTeamNameForEmail(recipient), currentUserTeam);    
 
             // Check if recipientDetails (recipientName or recipientTeam) is available.
-            if (recipientName == null || recipientTeam == null){
+            if (recipientName == null || recipientTeam == null) {
                 continue;  
             }   
 
-            String recipientFragmentHtml = updateTemplate(fragmentTemplateToUse, min, max, average, averageExcludingSelf, recipient, df, recipientTeam, recipientName, showAvgExcludingSelf);
+            String recipientFragmentHtml = updateTemplate(fragmentTemplateToUse, min, max, average,
+                    averageExcludingSelf, recipient, df, recipientTeam, recipientName, showAvgExcludingSelf);
             fragmentHtml.append(recipientFragmentHtml); 
         }           
         return fragmentHtml.toString();                                                       
@@ -579,13 +583,13 @@ FeedbackQuestionDetails {
 
         for (FeedbackResponseAttributes response : responses) {
             FeedbackNumericalScaleResponseDetails responseDetails =
-                (FeedbackNumericalScaleResponseDetails) response.getResponseDetails();
+                    (FeedbackNumericalScaleResponseDetails) response.getResponseDetails();
             double answer = responseDetails.getAnswer();
             String giverEmail = response.giver;
             String recipientEmail = response.recipient;
 
             // Compute number of responses including user's self response
-            numResponses.put(recipientEmail, (numResponses.getOrDefault(recipientEmail, 0) + 1));
+            numResponses.put(recipientEmail, numResponses.getOrDefault(recipientEmail, 0) + 1);
 
             // Compute number of responses excluding user's self response
             numResponsesExcludingSelf.putIfAbsent(recipientEmail, 0);
@@ -593,7 +597,7 @@ FeedbackQuestionDetails {
 
             // simplify?
             if (!isSelfResponse) {
-                numResponsesExcludingSelf.put(recipientEmail, (numResponsesExcludingSelf.get(recipientEmail) + 1));
+                numResponsesExcludingSelf.put(recipientEmail, numResponsesExcludingSelf.get(recipientEmail) + 1);
             }
 
             // Compute minimum score received
@@ -603,7 +607,7 @@ FeedbackQuestionDetails {
             max.put(recipientEmail, Math.max(answer, max.getOrDefault(recipientEmail, answer)));
 
             // Compute total score received
-            total.put(recipientEmail, (total.getOrDefault(recipientEmail, 0.0) + answer));
+            total.put(recipientEmail, total.getOrDefault(recipientEmail, 0.0) + answer);
 
             // Compute total score received excluding self
             totalExcludingSelf.putIfAbsent(recipientEmail, null);
@@ -617,13 +621,13 @@ FeedbackQuestionDetails {
             }
 
             // Compute average score received
-            average.put(recipientEmail, (total.get(recipientEmail) / numResponses.get(recipientEmail)));
+            average.put(recipientEmail, total.get(recipientEmail) / numResponses.get(recipientEmail));
 
             // Compute average score received excluding self
             averageExcludingSelf.putIfAbsent(recipientEmail, null);
             if (!isSelfResponse && totalExcludingSelf.get(recipientEmail) != null) {
-                averageExcludingSelf.put(recipientEmail, (totalExcludingSelf.get(recipientEmail) 
-                            / numResponsesExcludingSelf.get(recipientEmail)));
+                averageExcludingSelf.put(recipientEmail, totalExcludingSelf.get(recipientEmail) 
+                            / numResponsesExcludingSelf.get(recipientEmail));
             }
         }
     }
@@ -636,7 +640,9 @@ FeedbackQuestionDetails {
         FeedbackParticipantType type = question.recipientType;
 
         for (FeedbackResponseAttributes response : responses) {
-            if (!bundle.visibilityTable.containsKey(response.getId())) continue;
+            if (!bundle.visibilityTable.containsKey(response.getId())) {
+                continue;
+            }
             if (!bundle.visibilityTable.get(response.getId())[1]
                     && type != FeedbackParticipantType.SELF
                     && type != FeedbackParticipantType.NONE) {
@@ -691,7 +697,7 @@ FeedbackQuestionDetails {
     public boolean isChangesRequiresResponseDeletion(
             FeedbackQuestionDetails newDetails) {
         FeedbackNumericalScaleQuestionDetails newNumScaleDetails =
-            (FeedbackNumericalScaleQuestionDetails) newDetails;
+                (FeedbackNumericalScaleQuestionDetails) newDetails;
 
         return this.minScale != newNumScaleDetails.minScale
             || this.maxScale != newNumScaleDetails.maxScale
@@ -764,7 +770,7 @@ FeedbackQuestionDetails {
         List<String> errors = new ArrayList<>();
         for (FeedbackResponseAttributes response : responses) {
             FeedbackNumericalScaleResponseDetails frd =
-                (FeedbackNumericalScaleResponseDetails) response.getResponseDetails();
+                    (FeedbackNumericalScaleResponseDetails) response.getResponseDetails();
             if (frd.getAnswer() < minScale || frd.getAnswer() > maxScale) {
                 errors.add(frd.getAnswerString() + Const.FeedbackQuestion.NUMSCALE_ERROR_OUT_OF_RANGE
                         + "(min=" + minScale + ", max=" + maxScale + ")");

--- a/src/main/java/teammates/common/datatransfer/questions/FeedbackNumericalScaleQuestionDetails.java
+++ b/src/main/java/teammates/common/datatransfer/questions/FeedbackNumericalScaleQuestionDetails.java
@@ -286,12 +286,12 @@ public class FeedbackNumericalScaleQuestionDetails extends
                 isRecipientTypeTeam, currentUserTeam);
 
         Set<String> recipientSet = numResponses.keySet();
-        ArrayList<String> recipientList = new ArrayList<>();
+        List<String> recipientList = new ArrayList<>();
 
         String fragmentHtml = fragmentHtmlUpdate(recipientSet, hiddenRecipients,
-                recipientList, isRecipientTypeGeneral, isRecipientTypeTeam,
-                isRecipientTypeStudent, showAvgExcludingSelf, currentUserTeam, currentUserIdentifier,
-                fragmentTemplateToUse, bundle, averageExcludingSelf, numResponses, min, max, average, df);    
+                recipientList, isRecipientTypeStudent, showAvgExcludingSelf,
+                currentUserTeam, currentUserIdentifier, fragmentTemplateToUse, 
+                bundle, averageExcludingSelf, numResponses, min, max, average, df);
 
         if (fragmentHtml.length() == 0) {
             return "";
@@ -305,10 +305,10 @@ public class FeedbackNumericalScaleQuestionDetails extends
                 : FormTemplates.NUMSCALE_RESULT_STATS;
         return Templates.populateTemplate(templateToUse,
                 Slots.SUMMARY_TITLE, statsTitle,
-                Slots.STATS_FRAGMENTS, fragmentHtml.toString());
+                Slots.STATS_FRAGMENTS, fragmentHtml);
     }
 
-    private void updateRecipientList(ArrayList<String> recipientList,
+    private void updateRecipientList(List<String> recipientList,
             Set<String> recipientSet, String currentUserIdentifier) {
         boolean hasCurrentUserReceivedAnyResponse = recipientSet.contains(currentUserIdentifier);
 
@@ -341,7 +341,7 @@ public class FeedbackNumericalScaleQuestionDetails extends
         String averageScoreExcludingSelfText =
                 getAverageExcludingSelfText(showAvgExcludingSelf, df, averageScoreExcludingSelf);
 
-        String newTemplate = Templates.populateTemplate(
+        return Templates.populateTemplate(
                 currentTemplate,
                 Slots.RECIPIENT_TEAM, SanitizationHelper.sanitizeForHtml(recipientTeam),
                 Slots.RECIPIENT_NAME, SanitizationHelper.sanitizeForHtml(recipientName),
@@ -350,7 +350,6 @@ public class FeedbackNumericalScaleQuestionDetails extends
                 Slots.MIN, df.format(minScore),
                 Slots.AVERAGE_EXCLUDING_SELF_RESPONSE, averageScoreExcludingSelfText);
 
-        return newTemplate;
     }
 
     private String getDisplayableRecipientName(boolean isHiddenRecipient,
@@ -445,7 +444,7 @@ public class FeedbackNumericalScaleQuestionDetails extends
 
         boolean showAvgExcludingSelf = showAverageExcludingSelf(question, averageExcludingSelf);
 
-        DecimalFormat df = customDecimalFormat(0, 5, "down");
+        // DecimalFormat df = customDecimalFormat(0, 5, "down");
 
         String csvHeader = "Team, Recipient, Average, Minimum, Maximum"
                 + (showAvgExcludingSelf ? ", Average excluding self response" : "")
@@ -500,9 +499,9 @@ public class FeedbackNumericalScaleQuestionDetails extends
         df.setMinimumFractionDigits(min);
         df.setMaximumFractionDigits(max);
         //  roundingMode = roundingMode.toUpperCase()
-        if (roundingMode.equals(new String("down"))) {
+        if ("down".equals(roundingMode)) {
             df.setRoundingMode(RoundingMode.DOWN);
-        } else if (roundingMode.equals(new String("up"))) {
+        } else if ("up".equals(roundingMode)) {
             df.setRoundingMode(RoundingMode.UP);
         }
         return df;
@@ -510,8 +509,7 @@ public class FeedbackNumericalScaleQuestionDetails extends
 
 
     private String fragmentHtmlUpdate(Set<String> recipientSet, List<String> hiddenRecipients,
-            ArrayList<String> recipientList, boolean isRecipientTypeGeneral,
-            boolean isRecipientTypeTeam, boolean isRecipientTypeStudent, boolean showAvgExcludingSelf,
+            List<String> recipientList, boolean isRecipientTypeStudent, boolean showAvgExcludingSelf,
             String currentUserTeam, String currentUserIdentifier, String fragmentTemplateToUse, 
             FeedbackSessionResultsBundle bundle, Map<String, Double> averageExcludingSelf,
             Map<String, Integer> numResponses, Map<String, Double> min, Map<String, Double> max,

--- a/src/test/java/teammates/test/cases/datatransfer/FeedbackResponseDetailsTest.java
+++ b/src/test/java/teammates/test/cases/datatransfer/FeedbackResponseDetailsTest.java
@@ -6,6 +6,14 @@ import java.util.List;
 
 import org.testng.annotations.Test;
 
+import teammates.common.datatransfer.CourseRoster;
+import teammates.common.datatransfer.DataBundle;
+import teammates.common.datatransfer.FeedbackSessionResultsBundle;
+import teammates.common.datatransfer.attributes.FeedbackQuestionAttributes;
+import teammates.common.datatransfer.attributes.FeedbackResponseAttributes;
+import teammates.common.datatransfer.attributes.FeedbackSessionAttributes;
+import teammates.common.datatransfer.attributes.InstructorAttributes;
+import teammates.common.datatransfer.attributes.StudentAttributes;
 import teammates.common.datatransfer.questions.FeedbackConstantSumQuestionDetails;
 import teammates.common.datatransfer.questions.FeedbackConstantSumResponseDetails;
 import teammates.common.datatransfer.questions.FeedbackContributionQuestionDetails;
@@ -22,14 +30,6 @@ import teammates.common.datatransfer.questions.FeedbackRubricQuestionDetails;
 import teammates.common.datatransfer.questions.FeedbackRubricResponseDetails;
 import teammates.common.datatransfer.questions.FeedbackTextQuestionDetails;
 import teammates.common.datatransfer.questions.FeedbackTextResponseDetails;
-import teammates.common.datatransfer.FeedbackSessionResultsBundle;
-import teammates.common.datatransfer.attributes.FeedbackResponseAttributes;
-import teammates.common.datatransfer.attributes.InstructorAttributes;
-import teammates.common.datatransfer.attributes.StudentAttributes;
-import teammates.common.datatransfer.attributes.FeedbackQuestionAttributes;
-import teammates.common.datatransfer.attributes.FeedbackSessionAttributes;
-import teammates.common.datatransfer.CourseRoster;
-import teammates.common.datatransfer.DataBundle;
 import teammates.common.util.Const;
 import teammates.test.cases.BaseTestCase;
 
@@ -292,7 +292,7 @@ public class FeedbackResponseDetailsTest extends BaseTestCase {
     }
 
     @Test
-    public void testStudentQuestionResultsStatisticsHtml () {
+    public void testStudentQuestionResultsStatisticsHtml() {
 
         // Load databundle with NUMSCALE questions- and session-info
         DataBundle db = loadDataBundle("/FeedbackSessionQuestionTypeTest.json");
@@ -353,17 +353,17 @@ public class FeedbackResponseDetailsTest extends BaseTestCase {
          * Methods to test
          */
         String resHtml = fnsqd.getQuestionResultStatisticsHtml(
-            fraList,
-            fqa,
-            "student1InCourse1@gmail.tmt",
-            bundle,
-            "student"
+                fraList,
+                fqa,
+                "student1InCourse1@gmail.tmt",
+                bundle,
+                "student"
         );
 
         String resCsv = fnsqd.getQuestionResultStatisticsCsv(
-            fraList,
-            fqa,
-            bundle
+                fraList,
+                fqa,
+                bundle
         );
 
         /**
@@ -371,7 +371,8 @@ public class FeedbackResponseDetailsTest extends BaseTestCase {
          */
         assertEquals("", resHtml);
         assertEquals(
-            "Team, Recipient, Average, Minimum, Maximum"+ Const.EOL +"\"Team 1.1</td></div>'\"\"\",\"student1 In Course1</td></div>'\"\"\",3.5,3.5,3.5" + Const.EOL, 
-            resCsv);
+                "Team, Recipient, Average, Minimum, Maximum" + Const.EOL 
+                + "\"Team 1.1</td></div>'\"\"\",\"student1 In Course1</td></div>'\"\"\",3.5,3.5,3.5" + Const.EOL, 
+                resCsv);
     }
 }

